src/app/page.tsx

```diff
@@ -6,12 +6,33 @@
 import PageContainer from "@/components/PageContainer";
 import SearchBar from "@/components/Searchbar";
 import { useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import { datasetList } from "@/services/discovery/index.public";
+import { SearchedDataset } from "@/services/discovery/types/dataset.types";
+import RecentDatasets from "@/components/RecentDatasets";
 import aboutBackground from "../public/homepage-about-background.png";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faArrowUpRightFromSquare } from "@fortawesome/free-solid-svg-icons";
 
 const HomePage = () => {
   const queryParams = useSearchParams();
+  const [datasets, setDatasets] = useState<SearchedDataset[]>([]);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await datasetList({
+          limit: 200,
+          sort: "createdAt desc",
+        });
+        setDatasets(response.data.datasets);
+      } catch (error) {
+        console.error("Error fetching datasets", error);
```

Comment:
important: you could show the error as alert. Please check for `setAlert` in other components.